src/twisted/internet/iocpreactor/udp.py

```diff
@@ -167,7 +167,12 @@ def doRead(self):
         )
 
         if rc and rc != ERROR_IO_PENDING:
-            self.handleRead(rc, data, evt)
+            # If the error was not 0 or IO_PENDING then that means recvfrom() hit a
+            # failure condition. In this situation recvfrom() gives us our response
+            # right away and we don't need to wait for Windows to call the callback
+            # on our event. In fact, windows will not call it for us so we must call it
+            # ourselves manually
+            self.reactor.callLater(0, self.cbRead, rc, data, evt)
```

Comment:
Hmm. I love the big comment here but are there any re-ordering concerns with this `callLater`? It feels a bit odd to defer this in a way which might get called out of order; why not call it synchronously?

Reply:
Great question!
I had originally created this as a synchronous call, but some of my teammates pointed out that there might a potential for a tight infinite loop here so we intentionally added in the callLater to give the reactor a chance to butt in.

The infinite loop we were worried about:
- Synchronously call `cbRead()` on this line
- `cbRead()` calls the `doRead()` function (basically recursion with one function inbetween)
- `recvfrom()` has another early response
- Synchronously call `cbRead()` 
- Repeat forever

I'm very curious to hear your thoughts!